resolver/src/main/java/io/netty/resolver/CompositeNameResolver.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.resolver;
+
+import io.netty.util.concurrent.EventExecutor;
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.FutureListener;
+import io.netty.util.concurrent.Promise;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static io.netty.util.internal.ObjectUtil.*;
+
+/**
+ * A composite {@link SimpleNameResolver} that resolves a host name against a sequence of {@link NameResolver}s.
+ *
+ * In case of a failure, only the last one will be reported.
+ */
+public final class CompositeNameResolver<T> extends SimpleNameResolver<T> {
```

Comment:
Perhaps we could keep this even if you make /etc/hosts resolution into DNS name resolver :-)